src/extension.c

```diff
@@ -92,41 +105,14 @@ extension_event_trigger(PG_FUNCTION_ARGS)
 	if (!CALLED_AS_EVENT_TRIGGER(fcinfo))
 		elog(ERROR, "not fired by event trigger manager");
 
+	/* TODO: does this do anything useful? */
 	if (strcmp(trigdata->event, "ddl_command_end") == 0 &&
 		strcmp(trigdata->tag, "CREATE EXTENSION") == 0 &&
 		IsA(trigdata->parsetree, CreateExtensionStmt) &&
 		strcmp(((CreateExtensionStmt *) trigdata->parsetree)->extname, EXTENSION_NAME) == 0)
 	{
 		extstate = EXTENSION_STATE_CREATING;
 	}
-	else if (strcmp(trigdata->event, "ddl_command_start") == 0 &&
-			 strcmp(trigdata->tag, "DROP EXTENSION") == 0 &&
-			 IsA(trigdata->parsetree, DropStmt))
-	{
-		DropStmt   *stmt = (DropStmt *) trigdata->parsetree;
-		const char *extname = strVal(linitial(linitial(stmt->objects)));
-
-		if (strcmp(extname, EXTENSION_NAME) == 0)
-		{
-			extstate = EXTENSION_STATE_DROPPING;
-
-			/*
-			 * Save the transaction ID, so that we can avoid going from
-			 * INITIAL to CREATED in the transaction that issued DROP
-			 * EXTENSION.
-			 */
-			drop_transaction_id = GetCurrentTransactionId();
-
-			/*
-			 * Notify other backends that the extension was dropped. We do
-			 * this via the relcache invalidation mechanism in Postgres by
-			 * issuing an invalidation on a proxy table. Other backends will
-			 * see an invalidation event on the proxy table and then knows
-			 * they need to move back to INITIAL state.
-			 */
-			CacheInvalidateRelcacheByRelid(extension_proxy_oid);
```

Comment:
How's this handled with your changes? Don't we need other backends to invalidate their extension state anymore? Did you test how dropping now works with multiple simultaneous backends?